swarm/src/dial_opts.rs

```diff
@@ -323,3 +350,32 @@ pub enum PeerCondition {
     /// configured connection limits.
     Always,
 }
+
+bitflags::bitflags! {
+    /// The available conditions under which a new dialing attempt to
+    /// a known peer is initiated. These conditions can be combined.
+    ///
+    /// ```
+    /// # use libp2p_swarm::dial_opts::{DialOpts, DialCondition};
+    /// # use libp2p_identity::PeerId;
+    /// #
+    /// DialOpts::peer_id(PeerId::random())
+    ///    .dial_conditions(DialCondition::Disconnected | DialCondition::NotDialing)
+    ///    .build();
+    /// ```
```

Comment:
I'd like some more elaborate examples here. Specifically, we should explain how `|` and `&` operate:

- `DialCondition::Disconnected | DialCondition::NotDialing` means make a new connection, unless we already have one OR are currently dialing one. The outcome will be that we have at most one connection (if the in-progress dial succeeds).

- `DialCondition::Disconnected & DialCondition::NotDialing` will be `0` in regards to the bitflags, right? What does this mean semantically? Always make a connection, regardless of whether we are dialing / connected? That is equivalent to not specifying any conditions, right?